zipline/data/data_portal.py

```diff
@@ -906,6 +910,32 @@ def get_history_window(self, assets, end_dt, bar_count, frequency, field,
         else:
             raise ValueError("Invalid frequency: {0}".format(frequency))
 
+        # Round prices according to asset type. The number of decimal places
+        # used for rounding is constant for all equities, but varies by future
+        # contract according to tick size.
+        decimals = {}
+        for asset in df:
```

Comment:
Looping here is probably too costly, I'll have to figure out a better way to do this.